src/atomic.rs

```diff
@@ -784,9 +784,11 @@ impl<T> AsMut<T> for Owned<T> {
 #[derive(Debug)]
 pub struct Ptr<'scope, T: 'scope> {
     data: usize,
-    _marker: PhantomData<&'scope T>,
+    _marker: PhantomData<(&'scope (), *const T)>,
```

Comment:
What's the difference here?

Reply:
The raw ptr will make it non Send by default.

Reply:
3 lines bellows it's made Send again but only for T: Send.

Reply:
A `Ptr` is *not* a reference - it's more like a pointer that can be dereferenced (using `.defer()` or `.as_ref()`).

Basically:

1. `_marker: PhantomData<&'scope T>` says: *I'm a reference.*
2. `_marker: PhantomData<(&'scope (), *const T)>` says: *I'm a shared pointer to `T` tied to `'scope`.*

The second option correctly describes `Ptr`.